bindings/elixir/native/src/ckzg_wrap.c

```diff
@@ -0,0 +1,839 @@
+    if (settings == NULL) {
+        fclose(fp);
+        return make_error(env, ckzg_atoms.failed_alloc_settings_resource);
+    }
+
+    C_KZG_RET ret = load_trusted_setup_file(settings, fp, precompute);
+    fclose(fp);
+
+    if (ret != C_KZG_OK) {
+        enif_release_resource(settings);
+        return make_kzg_error(env, ret);
+    }
+
+    ERL_NIF_TERM settings_term = enif_make_resource(env, settings);
+    enif_release_resource(settings);
+
+    return make_success(env, settings_term);
+}
+
+static ERL_NIF_TERM blob_to_kzg_commitment_nif(
+    ErlNifEnv *env, int argc, const ERL_NIF_TERM argv[]
+) {
+    if (argc != 2) return make_error(env, ckzg_atoms.incorrect_arg_count);
+
+    ErlNifBinary blob;
+    if (!enif_inspect_binary(env, argv[0], &blob))
+        return make_error(env, ckzg_atoms.blob_not_binary);
+
+    if (blob.size != BYTES_PER_BLOB) return make_error(env, ckzg_atoms.invalid_blob_length);
+
+    KZGSettings *settings;
+    if (!enif_get_resource(env, argv[1], KZGSETTINGS_RES_TYPE, (void **)&settings))
+        return make_error(env, ckzg_atoms.failed_get_settings_resource);
+
+    ERL_NIF_TERM commitment_term;
+    unsigned char *commitment = enif_make_new_binary(env, BYTES_PER_COMMITMENT, &commitment_term);
+    if (commitment == NULL) return make_error(env, ckzg_atoms.out_of_memory);
+
+    C_KZG_RET ret = blob_to_kzg_commitment(
+        (KZGCommitment *)commitment, (Blob *)blob.data, settings
+    );
+    if (ret != C_KZG_OK) return make_kzg_error(env, ret);
+
+    return make_success(env, commitment_term);
+}
+
+static ERL_NIF_TERM compute_kzg_proof_nif(ErlNifEnv *env, int argc, const ERL_NIF_TERM argv[]) {
+    if (argc != 3) return make_error(env, ckzg_atoms.incorrect_arg_count);
+
+    ErlNifBinary blob;
+    if (!enif_inspect_binary(env, argv[0], &blob))
+        return make_error(env, ckzg_atoms.blob_not_binary);
+
+    if (blob.size != BYTES_PER_BLOB) return make_error(env, ckzg_atoms.invalid_blob_length);
+
+    ErlNifBinary z;
+    if (!enif_inspect_binary(env, argv[1], &z)) return make_error(env, ckzg_atoms.z_not_binary);
+
+    if (z.size != BYTES_PER_FIELD_ELEMENT) return make_error(env, ckzg_atoms.invalid_z_length);
+
+    KZGSettings *settings;
+    if (!enif_get_resource(env, argv[2], KZGSETTINGS_RES_TYPE, (void **)&settings))
+        return make_error(env, ckzg_atoms.failed_get_settings_resource);
+
+    ERL_NIF_TERM y_term;
+    unsigned char *y_bytes = enif_make_new_binary(env, sizeof(Bytes32), &y_term);
+    if (y_bytes == NULL) return make_error(env, ckzg_atoms.out_of_memory);
+
+    ERL_NIF_TERM proof_term;
+    unsigned char *proof_bytes = enif_make_new_binary(env, sizeof(KZGProof), &proof_term);
+    if (proof_bytes == NULL) return make_error(env, ckzg_atoms.out_of_memory);
+
+    C_KZG_RET ret = compute_kzg_proof(
+        (KZGProof *)proof_bytes, (Bytes32 *)y_bytes, (Blob *)blob.data, (Bytes32 *)z.data, settings
+    );
```

Comment:
Just a nit, I would probably add the `const` qualifier to the cast where applicable. This applies to other calls too.